.travis.yml

```diff
@@ -6,9 +6,17 @@ node_js:
     - "7"
     - "8"
 sudo: false
-script: "npm test && npm run docs"
+branches:
+    only:
+    - master
+
+# Run npm test always
+# Run npm docs only when its the master branch build and node 8
+script:
+    - "npm test"
+    - 'if [[ "$TRAVIS_PULL_REQUEST" = "false" && "$TRAVIS_BRANCH" = "master" && "$node_js" = "8" ]]; then npm run docs; fi'
```

Comment:
Agree. Our jsdoc generation is not even documented anywhere. To tell the truth, I've been working on ESLint since week 3, and I have never generated or even looked at our jsDoc output, so I'm not sure we should be re-running jsdoc generation on every build.